### PR TITLE
STRY4-TASK32: Data-is-stored-in-db

### PR DESCRIPTION
**Pull request by sanj06**
Validation for the incoming request with user data. 
Store encrypted user data in corresponding DB.
Send response back with user-id token.

### CODE REVIEW

File: backend/models/helper.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+}
+
+async function exists(queryObj, model) {
+    const data = await model.findOne(queryObj).lean();
+    return data;   
+}
+
+async function storeUser(email, password, name, model) {
+    //hash password
+    const salt = await bcrypt.genSalt(10);
```
Comment (and-le) [inline backend/models/helper.js:57]: Any particular reason for the choice of 10? This looks good, just curious.

**Review by and-le — APPROVED** [review]
Looks good to me.

**Comment by and-le** [comment]
Can you @sanj06 update the description to close the appropriate tasks before merging?

File: backend/models/helper.js (hunk 1 above)
Comment (s1hu) [reply backend/models/helper.js:57]: 10 is the default value for genSalt(), meaning there will be 10 rounds for salting. Too much salting will overcomplicate the hash algorithm and too little will be insecure. For our server, using the default might be the best choice

**Review by s1hu — APPROVED** [review]
Reviewed implementation for login and signup in backend.

File: backend/models/helper.js (hunk 1 above)
Comment (sanj06) [reply backend/models/helper.js:57]: I think it shouldn't be too big that it takes too much time to hash/unhash nor too small that it's easy to get hacked - I believe 10 is the default option and is somewhere in the middle

**Pull request merged**